plugins/slicer/MONAILabelReviewer/MONAILabelReviewer.py

```diff
@@ -0,0 +1,1070 @@
+        self.parent.helpText = """
+This module provides the user to review on segmentations on X-Ray-dicom images.
+See more information in <a href="...">module documentation</a>.
+"""
+        self.parent.acknowledgementText = """
+Developed by rAiDiance, and  funded by Berlin Institute of Health (BIH).
+"""
+
+
+class MONAILabelReviewerWidget(ScriptedLoadableModuleWidget, VTKObservationMixin):
+    """Uses ScriptedLoadableModuleWidget base class, available at:
+    https://github.com/Slicer/Slicer/blob/master/Base/Python/slicer/ScriptedLoadableModule.py
+    """
+
+    def __init__(self, parent=None):
+        """
+        Called when the user opens the module the first time and the widget is initialized.
+        """
+        ScriptedLoadableModuleWidget.__init__(self, parent)
+        VTKObservationMixin.__init__(self)  # needed for parameter node observation
+        self.logic = None
+        self._parameterNode = None
+        self._updatingGUIFromParameterNode = False
+
+        self.STATUS = SegStatus()
+        self.LEVEL = Level()
+
+        self.selectedReviewer: str = ""
+        self.selectedClientId: str = ""
+        self.listImageData: List[ImageData] = None
+        self.imageCounter: int = 0
+        self.currentImageData: ImageData = None
+        self.idToimageData: Dict[str, ImageData] = None
+
+        # Meta Information
+        self.finalStatus: str = ""
+        self.finalLevel: str = ""
+        self.finalComment: str = ""
+
+        self.reviewersModeIsActive = False
+
+        self.mapFiltersToBool: Dict[str, bool] = {
+            "segmented": False,
+            "notSegemented": False,
+            "approved": False,
+            "flagged": False,
+        }
+
+    def setup(self):
+        """
+        Called when the user opens the module the first time and the widget is initialized.
+        """
+        ScriptedLoadableModuleWidget.setup(self)
+
+        # Load widget from .ui file (created by Qt Designer).
+        # Additional widgets can be instantiated manually and added to self.layout.
+        uiWidget = slicer.util.loadUI(self.resourcePath("UI/MONAILabelReviewer.ui"))
+        self.layout.addWidget(uiWidget)
+        self.ui = slicer.util.childWidgetVariables(uiWidget)
+
+        self.dialogWidget = slicer.util.loadUI(self.resourcePath("UI/DialogInfo.ui"))
+        # Set scene in MRML widgets. Make sure that in Qt designer the top-level qMRMLWidget's
+        # "mrmlSceneChanged(vtkMRMLScene*)" signal in is connected to each MRML widget's.
+        # "setMRMLScene(vtkMRMLScene*)" slot.
+        uiWidget.setMRMLScene(slicer.mrmlScene)
+
+        # Create logic class. Logic implements all computations that should be possible to run
+        # in batch mode, without a graphical user interface.
+        self.logic = MONAILabelReviewerLogic()
+
+        self.setLightVersion()
+        self.segmentEditorWidget = qSlicerSegmentationsModuleWidgetsPythonQt.qMRMLSegmentEditorWidget()
```

Comment:
@lassoan Thoughts on this type of initialization method for the segment editor widget. Cc’ing based on your recent improvements made in https://github.com/Project-MONAI/MONAILabel/commit/832f8c2d2c711d1ec84a3e6338d66d1230f9e6ae.

Reply:
Having a single segment editor widget instance is good.

It is not guaranteed that `qMRMLSegmentEditorWidget` will forever be in `qSlicerSegmentationsModuleWidgetsPythonQt` module, so I would recommend to get it from the `slicer` package (`self.segmentEditorWidget = slicer.qMRMLSegmentEditorWidget()`).

The scene and parameter nodes are set correctly later, so that's good, too. However, it may be simpler to set them both here right away and then you don't need to deal with them later. It is safe to set them here because by the time module widgets are created, the scene is already set up. You don't need to worry about scene closing, because you use a singleton node as parameter node; and singleton nodes are just reset to default (not deleted) when the scene is closed.

Reply:
Done
Thanks a lot for the advice. I have improved it accordingly. 
(qMRMLSegmentEditorWidget was automatically included when setting up the extension template which was provided by slicer)